### PR TITLE
Linux testing

### PR DESCRIPTION
**Pull request by jkingoliver**
@schen22 This is a workaround for the Bundle errors we see in Linux.  It's for the Dialog service, which I didn't realize we don't need to support on Linux since it's deprecated. Let me know if you want to merge this or not.

### CODE REVIEW

**Comment by rfdickerson** [comment]
Hi @jkingoliver - is this the latest? Should @schen22 merge?

**Comment by jkingoliver** [comment]
@schen22, @rfdickerson 
This is now ready for a squash and merge, after review by Sarah.

**Comment by schen22** [comment]
Great - I'll look into it as soon as I can and hopefully have it in our next release with the new Discovery service we've been working hard to support. Thanks @jkingoliver!

**Comment by germanattanasio** [comment]
Most of the commits here were made from an account that wasn't configured correctly. Can we fix that ?

**Comment by germanattanasio** [comment]
We may have to close an open a new PR once the github config is updated. Otherwise you won't be able to sign the CLA.

**Comment by rfdickerson** [comment]
I'm wondering, should we update the README at all to mention basic support for Linux? Also, it looks like the Travis CI script seems to be in conflict at this point.

**Comment by schen22** [comment]
It'd be great if the README is updated to mention support for Linux! The Travis script is failing because we released support for the new Discovery service right before leaving to vacation.

File: .travis.yml [hunk 1]
```diff
@@ -1,33 +1,47 @@
-language: objective-c
-osx_image: xcode8
-branches:
-  only:
-    - master
```
Comment (schen22) [inline .travis.yml]: What's the reason for removing the specification for `master` branch in the build setting?

**Comment by schen22** [comment]
Just finished running and fixing all the tests that were failing previously. Once a new PR is created and @jkingoliver signs the CLA with these changes, I can merge it into master. 

Thanks Jeannie for updating our variable names, removing Foundation dependencies and force unwraps in our tests! It's awesome how these changes allow us to support Linux :)

**Comment by jkingoliver** [comment]
I've signed the CLA. @schen22 There is a conflict with the travis.yml that needs to be resolved still. I'm not sure if I'll need to create a new PR or not, we'll see after the conflict is resolved.

As far as updating the README with Linux support, we should make it clear that it's still Beta because of the unresolved issues in Swift-Runtime:

1. #259: We can't currently include message bodies in the request or response larger than ~16Kb, which affects these services:
Alchemy Language
Document Conversion
Text to Speech
Visual Recognition

2. #242 We can't include some characters in query parameters, such as whitespace and `^`. This affects:
Alchemy Data News 
Alchemy Language

The services that we can support after this PR, i.e. testcases are passing, are:
Conversation
Language Translator
Natural Language Classifier
Personality Insights V3
Tone Analyzer
Tradeoff Analytics

**Comment by jkingoliver** [comment]
I think it's a reasonable question whether we want to claim basic support now and call out these major problems or hold off on Linux support until they're fixed. Thoughts?

**Comment by schen22** [comment]
I can resolve the conflicts for the Travis script! Before I do so though, could you tell me why you removed the part specifying master branch from the script? 

```
-branches:
-  only:
-    - master 
```

As for the README, @rfdickerson  and I were discussing yesterday we might not want to mention the Swift-Runtime issues because they're Apple dependent and a change within Foundation is needed before they can be resolved. For now, you can remove the Linux badge I added until we can support all the services and update the README to include that Linux offers support for Conversation, Language Translator, Natural Language Classifier, Personality Insights V3, Tone Analyzer, and Tradeoff Analytics, and add your plans and timeline for when support for the rest of the services will be added. What do you think?

Also, you might have linked the wrong issues on accident too -- issues 259 and 242 are both closed within the SDK and don't refer to Linux testing.

**Comment by jkingoliver** [comment]
The change in the travis file was made as a temporary measure to see if my branch would build.  I'll add that back in.

I like your suggestion to announce support for those specific services.

Those issues I mentioned above are in the Swift-Runtime repo, so they won't be accurately linked from here.

**Comment by jkingoliver** [comment]
@schen22 I have updated the platform badge, the section on Linux compatibility and the travis file.  Will you take a look please?

**Comment by schen22** [comment]
Looks great, thanks!

**Pull request merged**